test/serializer/optimizations/residualDeadCode.js

```diff
@@ -0,0 +1,17 @@
+// does not contain:eliminate
+(function () {
+  let f = false;
+  let t = true;
+  global.inspect = function () {
+    if (f) { console.log("eliminate me"); }
+    if (t) { } else { console.log("eliminate me"); }
```

Comment:
I see three cases in your handling of the IfStatement above, but only two are getting exercised here.

Reply:
Added a test. I'm not super happy with these tests, though - there doesn't seem to be a good way of asserting the precise shape of the output. For example, I can assert that `[] || 'eliminate'` prepacks into something not containing `'eliminate'`, but not that it actually ends up as `[]`, which is an important property to preserve.

Have you considered adding snapshot tests? For example, the `prettier` project uses them [almost exclusively](https://github.com/prettier/prettier/tree/master/tests), to great effect. It makes it easy to see what the impact of a change is over a large range of code.